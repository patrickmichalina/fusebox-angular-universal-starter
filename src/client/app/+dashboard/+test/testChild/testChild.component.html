<mat-card>
    Test Child Project
</mat-card>