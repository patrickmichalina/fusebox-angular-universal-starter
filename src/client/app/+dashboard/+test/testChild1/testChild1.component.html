<md-card>
  Test Child 1 Project
</md-card>