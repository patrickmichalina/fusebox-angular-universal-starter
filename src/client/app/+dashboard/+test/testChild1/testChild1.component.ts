import { ChangeDetectionStrategy, Component } from '@angular/core'

@Component({
  selector: 'pm-test-child-2',
  templateUrl: './testChild1.component.html',
  styleUrls: ['./testChild1.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class TestChild1Component {
}
