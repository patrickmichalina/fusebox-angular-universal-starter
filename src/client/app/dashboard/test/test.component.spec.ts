import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { Component } from '@angular/core'
import { TestModule } from './test.module'
import { TestComponent } from './test.component'
import { RouterTestingModule } from '@angular/router/testing'
import { TestChildComponent } from './testChild/testChild.component'
import { TestChild1Component } from './testChild1/testChild1.component'
import { Route } from '@angular/router'
import { EnvironmentService } from '../../shared/services/environment.service'
import { APP_BASE_HREF } from '@angular/common'

describe(TestComponent.name, () => {
  const config: Array<Route> = [
    {
      path: '',
      component: TestComponent,
      pathMatch: 'full',
      children: [
        { path: '', redirectTo: 'test-child', pathMatch: 'full' },
        {
          path: 'test-child',
          component: TestChildComponent
        },
        {
          path: 'test-child-1',
          component: TestChild1Component
        }
      ]
    }
  ]

  let fixture: ComponentFixture<TestComponent>

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [TestModule, RouterTestingModule.withRoutes(config)],
      declarations: [TestingComponent],
      providers: [
        { provide: APP_BASE_HREF, useValue: '/' },
        EnvironmentService
      ]
    }).compileComponents()
  }))

  beforeEach(async(() => {
    fixture = TestBed.createComponent(TestComponent)
  }))

  afterEach(async(() => {
    TestBed.resetTestingModule()
  }))

  it('should match snapshot', async(() => {
    expect(fixture).toMatchSnapshot()
  }))

  it('should compile', async(() => {
    expect(fixture.nativeElement).toBeDefined()
  }))
})

@Component({
  selector: 'test-component',
  template: '<pm-test></pm-test>'
})
class TestingComponent {}
