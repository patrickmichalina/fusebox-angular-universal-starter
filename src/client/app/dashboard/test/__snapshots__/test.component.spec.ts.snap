// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent should match snapshot 1`] = `
<pm-test>
  <router-outlet />
</pm-test>
`;
